test/Microsoft.Dnx.Testing.CommonUtils/Directory/Dir.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Dnx.Testing
+{
+    public class Dir
+    {
+        public const string EmptyFile = "";
+        private readonly Dictionary<string, object> _nodes = new Dictionary<string, object>();
+
+        public Dir(string rootPath): this()
+        {
+            Load(rootPath);
+        }
+
+        public Dir()
+        {
+            ReadFile = fileInfo => File.ReadAllText(fileInfo.FullName);
+        }
+
+        public Dir(Dir parent)
+        {
+            ReadFile = parent.ReadFile;
+        }
+
+        public Func<FileInfo, object> ReadFile;
+
+        public string LoadPath { get; private set; } = "In Memory";
+
+        public IEnumerable<KeyValuePair<string, object>> Nodes
+        {
+            get
+            {
+                return _nodes;
+            }
+        }
+
+        public object this[params string[] keys]
+        {
+            set
+            {
+                foreach (var key in keys)
+                {
+                    _nodes[key] = value;
+                }
+            }
+        }
+
+        public void Load(string path)
+        {
+            LoadPath = path;
+            var directory = new DirectoryInfo(path);
+
+            foreach (var subDirectoryInfo in directory.EnumerateDirectories())
+            {
+                Read(this, subDirectoryInfo);
+            }
+
+            foreach (var file in directory.EnumerateFiles())
+            {
+                this[file.Name] = ReadFile(file);
+            }
+        }
+
+        public void Save(string path)
+        {
+            Write(path, this);
+        }
+
+        private void Write(string path, object value)
+        {
+            var tree = value as Dir;
+            if (tree != null)
+            {
+                Directory.CreateDirectory(path);
+
+                foreach (var node in tree.Nodes)
+                {
+                    Write(Path.Combine(path, node.Key), node.Value);
+                }
+            }
+            else
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(path));
+                File.WriteAllText(path, value?.ToString() ?? string.Empty);
+            }
+        }
+
+        private static void Read(Dir parent, DirectoryInfo directoryInfo)
+        {
+            var directory = new Dir(parent);
```

Comment:
There's a lot of duplicated code here and in `Load`. Can we just use `Read` directly in `Load`?